proto/rill/runtime/v1/resources.proto

```diff
@@ -301,6 +302,27 @@ message BucketExtractPolicy {
   uint64 files_limit = 4;
 }
 
+message Theme {
+  ThemeSpec spec = 1;
+  ThemeState state = 2;
+}
+
+message ThemeSpec {
+  message Color {
+    float red = 1;
+    float green = 2;
+    float blue = 3;
+    float alpha = 4;
+  }
+  message Colors {
+    optional Color primary = 1;
+    optional Color secondary = 2;
+  }
+  Colors colors = 1;
```

Comment:
Would recommend flattening here (not in the YAML though), to just `primary_color` and `secondary_color`. Flattening is nice in the protobufs since it reduces the number of types.